src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -36,15 +36,6 @@ function AxisSystem(props: Props): JSX.Element {
     ordinateScale.invert(height / (2 * zoom) + position.y),
   ];
 
-  // Restrain ticks scales to visible domains
-  const xTicksScale = abscissaInfo.scaleFn();
-  xTicksScale.domain(xVisibleDomain);
-  xTicksScale.range([0, width]);
-
-  const yTicksScale = ordinateInfo.scaleFn();
-  yTicksScale.domain(yVisibleDomain);
-  yTicksScale.range([height, 0]);
-
```

Comment:
Scale construction was moved inside `Axis` to:
- Avoid a fallback of `scaleType` in `AxisSystem` **and** `Axis`
- Avoid passing `scale` to `Axis` as it already depends on `domain` and `canvasSize`

I had to introduce a new flag `flipAxis` to deal with the inverted range of yAxis but I think it might be handy later.

Reply:
Nice! I like it 💯